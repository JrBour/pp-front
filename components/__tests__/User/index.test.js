import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import User from '@/components/User'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('User', () => {
  let wrapper
  let mutations
  let store
  const removeParticipantMock = jest.fn()

  beforeEach(() => {
    mutations = {
      removeParticipant: removeParticipantMock
    }
    
    store = new Vuex.Store({ mutations })
    wrapper = shallowMount(User, {
      localVue,
      store,
      propsData: {
        user: {
          id: '12',
          givenName: 'John',
          lastName: 'Doe',
          image: {
            filePath: 'pp.jpg'
          }
        },
      },
    })
  })

  describe('methods', () => {
    it('should emitted removeParticipants methods', async () => {
      await wrapper.find('button').trigger('click')
      expect(removeParticipantMock.mock.calls.length).toBe(1)
    })
  })

  describe('snapshot', () => {
    it('renders properly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
