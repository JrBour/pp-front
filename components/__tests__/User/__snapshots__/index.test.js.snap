// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User snapshot renders properly 1`] = `
<div class="user">
  <div class="user__wrapper">
    <div class="user__image" style="background-image: url(undefined/media/pp.jpg); background-repeat: no-repeat; background-size: cover; background-position: center;"></div> <button>
      <p>x</p>
    </button>
  </div>
  <p><span>John</span> <br>
    Doe
  </p>
</div>
`;
